From 518fa756bab9b107bc94b392dbbf8cb23a607cb8 Mon Sep 17 00:00:00 2001
Message-Id: <518fa756bab9b107bc94b392dbbf8cb23a607cb8.1436851672.git.chang-joon.lee@intel.com>
From: vandita kulkarni <vandita.kulkarni@intel.com>
Date: Mon, 6 Jul 2015 20:39:25 +0530
Subject: [PATCH] REVERTME [VPG]: drm/i915: Enable PM5 only during single plane
 state

To prevent race conditions between multi plane exit and PM5 entry
scenarios, wait for the display controller to enter into single plane
mode and then enable PM5. Otherwise flicker or blankout happens due
to pipe underruns.

REVERTME: This patch depends on vpg-atomicity design
and needs to be reverted when nuclear flip design
from upstream gets in.

Change-Id: Ic58e147d7b06214224ad11fdb7b682bcd78a83d9
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-28028
Signed-off-by: vandita kulkarni <vandita.kulkarni@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index f9021aa..0624988 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -1686,7 +1686,8 @@ void intel_update_maxfifo(struct drm_i915_private *dev_priv,
 
 	if (enable) {
 		if (IS_CHERRYVIEW(dev_priv->dev)) {
-
+			intel_wait_for_vblank(dev_priv->dev,
+					(to_intel_crtc(crtc))->pipe);
 			/*
 			 * When display is in single plane maxfifo mode,
 			 * DDR DVFS can be enabled for better power saving.
-- 
1.9.1

